src/ArrangerConduit.sol

```diff
@@ -0,0 +1,256 @@
+
+        delete fundRequests[fundRequestId];
+
+        requestedFunds[ilk][asset] -= amountRequested;
+        totalRequestedFunds[asset] -= amountRequested;
+
+        emit CancelFundRequest(fundRequestId);
+    }
+
+    /**********************************************************************************************/
+    /*** Fund Manager Functions                                                                 ***/
+    /**********************************************************************************************/
+
+    function drawFunds(address asset, uint256 amount) external override isArranger {
+        require(amount <= drawableFunds(asset), "Conduit/insufficient-funds");
+
+        require(ERC20Like(asset).transfer(arranger, amount), "Conduit/transfer-failed");
+
+        emit DrawFunds(asset, amount);
+    }
+
+    function returnFunds(uint256 fundRequestId, uint256 returnAmount)
+        external override isArranger
+    {
+        FundRequest storage fundRequest = fundRequests[fundRequestId];
+
+        require(fundRequest.status == StatusEnum.PENDING, "Conduit/invalid-status");
+
+        address asset = fundRequest.asset;
+        bytes32 ilk   = fundRequest.ilk;
+
+        uint256 amountRequested = fundRequest.amountRequested;
+
+        withdrawableFunds[ilk][asset] += returnAmount;
+        totalWithdrawableFunds[asset] += returnAmount;
+
+        requestedFunds[ilk][asset] -= amountRequested;
+        totalRequestedFunds[asset] -= amountRequested;
+
+        fundRequest.amountFilled += returnAmount;
+
+        fundRequest.status = StatusEnum.COMPLETED;
+
+        require(
+            ERC20Like(fundRequest.asset).transferFrom(arranger, address(this), returnAmount),
+            "Conduit/transfer-failed"
+        );
+
+        emit ReturnFunds(ilk, asset, fundRequestId, amountRequested, returnAmount);
+    }
+
+    /**********************************************************************************************/
+    /*** View Functions                                                                         ***/
+    /**********************************************************************************************/
+
+    function drawableFunds(address asset) public view returns (uint256 drawableFunds_) {
+        drawableFunds_ = ERC20Like(asset).balanceOf(address(this)) - totalWithdrawableFunds[asset];
+    }
+
+    function maxDeposit(bytes32 ilk, address asset)
+        external override pure returns (uint256 maxDeposit_)
+    {
+        ilk; asset;  // Silence warnings
+        maxDeposit_ = type(uint256).max;
+    }
+
+    function maxWithdraw(bytes32 ilk, address asset)
+        external override view returns (uint256 maxWithdraw_)
+    {
+        maxWithdraw_ = withdrawableFunds[ilk][asset];
+    }
+
+    function isCancelable(uint256 fundRequestId)
+        external override view returns (bool isCancelable_)
+    {
+        isCancelable_ = fundRequests[fundRequestId].status == StatusEnum.PENDING;
```

Comment:
This will incorrectly return `true` if the user specifies an unused `fundRequestId` as `StatusEnum.PENDING == 0`. I think it makes sense to add an additional status to capture this in all cases:

```
enum StatusEnum {
    UNINITIALIZED,
    PENDING,
    CANCELLED,
    COMPLETED
 }
```